File: code/cpp/945.cpp
//https://leetcode.com/problems/maximum-product-after-k-increments/
class Solution {
public:
    int minIncrementForUnique(vector<int>& nums) {
        size_t res = 0, shift = 0;
        sort(nums.begin(), nums.end());
        for (size_t i = 1; i < nums.size(); i++) {
            if (nums[i] <= nums[i - 1]) {
                shift = nums[i - 1] - nums[i] + 1;
                nums[i] += shift;
                res += shift;
            }
        }
        return res;
    }
};

File: code/python/945.py
#https://leetcode.com/problems/maximum-product-after-k-increments/
class Solution:
    def minIncrementForUnique(self, nums: List[int]) -> int:
        res = shift = 0
        nums.sort()
        for i in range(1, len(nums)):
            if nums[i] <= nums[i - 1]:
                shift = nums[i - 1] - nums[i] + 1
                nums[i] += shift
                res += shift
        return res
